prime-router/src/main/kotlin/azure/ReportFunction.kt

```diff
@@ -175,7 +168,19 @@ class ReportFunction : Logging {
             val options = Options.valueOf(optionsText)
 
             // track the sending organization and client based on the header
-            val validatedRequest = validateRequest(workflowEngine, request)
+            val validatedRequest = validateRequest(request)
+            val rawBody = validatedRequest.content.toByteArray()
+            // TODO this should be only calculated once and passed, but the underlying functions are called by
+            //  receive (this), process, batch, send and those do *not* need to calculate it outside of the function
+            //  so leaving it 2x calculating on receive for the time being.
+            val digest = BlobAccess.sha256Digest(rawBody)
```

Comment:
if these are only used if `sender.allowDuplicates` is true, move them into the conditional

Reply:
Good call. updated.